op-batcher/batcher/channel_builder.go

```diff
@@ -337,6 +338,15 @@ func (c *channelBuilder) outputFrame() error {
 	return err // possibly io.EOF (last frame)
 }
 
+// Close immediately marks the channel as full with an ErrTerminated
+// if the channel is not already full, then outputs any remaining frames.
+func (c *channelBuilder) Close() error {
+	if !c.IsFull() {
+		c.setFullErr(ErrTerminated)
+	}
+	return c.closeAndOutputAllFrames()
```

Comment:
I think you might want to move `c.closeAndOutputAllFrames()` to inside the not full if statement. It will return an error here b/c the channel shouldn't be closed twice.

Reply:
The goal was to make a call to `ChannelBuilder.Close` emit any remaining frames and prevent any future frames from being outputted.

Another option would be to have `ChannelBuilder.Close` only call `setFullErr` (if unset) and leave the responsibility of outputting remaining frames to the standard `OutputFrames` method. If we take this approach, then we will probably want to cease calling `ChannelManager.processBlocks()` once the manager has been closed in order to ensure a timely shutdown. This alternative approach would probably help us to avoid double-closing the channel builder, making this potentially a safer route.